### ISSUE TITLE
'melt color:red' Does not work in v6.16.0: No Video & [consumer sdl2] audio timed out

### ISSUE THREAD

**Issue by RobertElder** [mltframework/mlt#445]
Hi, my high-level goal is to edit a short video in Kdenlive on Ubuntu 18.04, but Kdenlive keeps telling me that the project files are 'Invalid' after open saved project files.  I am filing a bug here because I believe I have found an mlt bug that may be contributing to this issue.  I have tested many versions of kdenlive including the latest version:

robert@robert-msi-ge62:~$ md5sum kdenlive-19.04.1b-x86_64.appimage 
f0ab2ecf3f606e1f1fc36e2a3929f9f7  kdenlive-19.04.1b-x86_64.appimage

Following the troubleshooting guide for kdenlive, I have tested out a few cases with melt:

https://kdenlive.org/en/troubleshooting/

### Steps taken to reproduce the bug in melt:

mkdir /tmp/a
cd /tmp/a
git clone https://github.com/mltframework/mlt.git
cd mlt
#  Release v6.16.0
git checkout 02bc879
export CXXFLAGS=-std=c++11
./configure --enable-gpl --enable-gpl3
make > saved_output 2>&1
. setenv
melt color:red   #  Window pops up with no red on it.  Doesn't close when you press 'x'.
                 #  After using ctrl+c, I see the message '[consumer sdl2] audio timed out'

#  Then, later I can pass a video through melt just fine:

melt /mnt/video/Screencast\ 2019-05-19\ 09:11:23.mp4 -consumer avformat:out.mp4 acodec=libmp3lame vcodec=libx264
+-----+ +-----+ +-----+ +-----+ +-----+ +-----+ +-----+ +-----+ +-----+
|1=-10| |2= -5| |3= -2| |4= -1| |5=  0| |6=  1| |7=  2| |8=  5| |9= 10|
+-----+ +-----+ +-----+ +-----+ +-----+ +-----+ +-----+ +-----+ +-----+
+---------------------------------------------------------------------+
|               H = back 1 minute,  L = forward 1 minute              |
|                 h = previous frame,  l = next frame                 |
|           g = start of clip, j = next clip, k = previous clip       |
|                0 = restart, q = quit, space = play                  |
+---------------------------------------------------------------------+
[mp4 @ 0x7fe2e4200ee0] Using AVStream.codec to pass codec parameters to muxers is deprecated, use AVStream.codecpar instead.
[mp4 @ 0x7fe2e4200ee0] Using AVStream.codec to pass codec parameters to muxers is deprecated, use AVStream.codecpar instead.

#  The resulting file 'out.mp4' plays just fine in VLC media player.

robert@robert-msi-ge62:~$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 18.04.2 LTS
Release:	18.04
Codename:	bionic

Other things to note:

- I have been able to play all these videos with ffplay.
- The offending video can be imported into kdenlive, edited, played, and cut up a bit, and then re-load fine.  Then, after editing with it more, the next time I re-open the kdenlive project, I'll get a message from kdenlive that says 'Invalid clip producer3 found on track playlist6 at 0. Invalid clip ...<repeated once for each up up part of the video>'.
-  After I get the 'Invalid' message, I can just re-import the same video and work with it just fine, but then I have to re-do the entire project every time I open a saved kdenlive project.
- The files that seem to cause this problem are ones I recoreded with 'kazam' a screen recording program.  Other videos I've included in the project that were recoreded on my phone will re-load properly in the same project while the 'Invalid' ones don't and are automatically removed from the kdenlive project.
- I installed a number of dependences that are mentioned in configure thinking that perhaps that some 'optional' dependency was the problem.
- I also tested this on commit f6225b7f4cd5e9b6011ca79c3849dc9e286a7acb 'Fix crop filter not working with color producer.', but got the same result.

Please advise on what further information you'd like or tests you'd like me to perform.

I was really just hoping to get a quick video edited, but I've spent several days installing kdenlive/mlt from source to try and figure out a build combination that works for standard Ubuntu.  This bug is also bad because it happen randomly after I've already invested an hour or so into working on the project.  I thought I got rid of it a couple times by switching to a new build, but it still keeps happening.

I am prepared to do a bit of work to figure this out, but I'm not too familiar with the dependency graph of mlt and kdenlive, so please advise where I should be looking.  If you want me to compile a different branch with some special printf statements and return the result, I can do that.  All other video editors for Linux suck, so I don't see any alternative than to get kdenlive working.

**I have attached several files including:**

-  ./configure output
-  make output
-  A screenshot of the window not being red.

![bug](https://user-images.githubusercontent.com/755903/57990140-8d2de880-7a71-11e9-8951-ec716e6c633d.png)
[configure-output.txt](https://github.com/mltframework/mlt/files/3195798/configure-output.txt)

[make_output.txt](https://github.com/mltframework/mlt/files/3195800/make_output.txt)

**Comment by ddennedy** [comment]
`melt color:red` works for me and nearly every one else. It is true that `melt` will not close when clicking the window close button when built against SDL2. When melt runs it prints keyboard usage help, and you can press `q` to quit. I just made a fix for the window manager close button. I see your `Current Position` has incremented. So, it is doing something, and perhaps SDL2 is not working good with your system. I think it defaults to OpenGL, and maybe running with environment `LIBGL_ALWAYS_SOFTWARE=1` will help.

With regard to MLT's compatibility with Kazam output, I ran a test using its H.264 output, and it works fine for me. It produces an extremely common H.264 MP4 in constant framerate and with no B frames.

As for kdenlive reporting errors while loading an XML project file, you need to take that up with them. I am not a kdenlive developer, and it is not shown/known to be a MLT problem. From the description, some editing operation is producing a bad result in the project file, and maybe you should focus more on the steps that lead to that. I strongly disagree that your melt SDL2 video problem is "an mlt bug that may be contributing to this issue."

**Comment by RobertElder** [comment]
Hi Dan,

Thank you for your quick response.  I have a few follow-up questions/comments:

-> "perhaps SDL2 is not working good with your system."

Do you have any suggestions on things I could test to determine if SDL2 problems are in fact the root cause as far as my kdenlive bug is concerned?  I can see on their site that that SDL2 is 'Simple DirectMedia Layer is a cross-platform development library designed to provide low level access to audio, keyboard, mouse, joystick, and graphics hardware via OpenGL and Direct3D. It is used by video playback software, emulators', but I'm not very familiar with this software bundle and how it fits together in this overall system.

I just tried doing:

export LIBGL_ALWAYS_SOFTWARE=1
. setenv
melt color:red

And I still see no color red.  Also, no video when playing a .mp4.

-> 'I strongly disagree that your melt SDL2 video problem is "an mlt bug that may be contributing to this issue."' 

That's good enough for me.  I suspected that my mlt issues might be unrelated, but since the troubleshooting guide for kdenlive suggests doing this simple mlt test, I figured I ought to do due diligence in trying to fix that first.  I will file a second bug for kdenlive with repo steps and cite this thread as evidence that the bug isn't further down in mlt.

In the event that I get pushback from the kdenlive bug report, do you have any further suggestions on how I might get mlt video and color output working if they ask me to do this first?  Also, should I be concerned about the message '[consumer sdl2] audio timed out'?  What approach would be used to investigate such a timeout?

Thanks,
Robert

**Comment by RobertElder** [comment]
I did more digging into why the the clips seem to become 'invalid' and I found that in the 'corrupted' .kdenlive project files, the track times seem to get changed by about a second or so.  This happens even if I open an existing project, make 0 changes then 'save as' another project file name.  If I then diff the two project files, the track times are different by some small margin.  I have no idea if that could be related to mlt, but I have filed a bug on the kdenlive bug tracker which I'll keep an eye on:

https://bugs.kde.org/show_bug.cgi?id=407778
